### PR TITLE
[FEATURE] Use parallel-lint for command ci:php:lint

### PR DESCRIPTION
**Pull request by Starkmann**
Fixes: #1139

### CODE REVIEW

**Review by oliverklee — CHANGES_REQUESTED** [review]
I propose we re-use the existing `ci:php:lint` Composer script. We then also need to add `composer install` and composer caching to the corresponding GitHub CI job.

File: composer.json [hunk 1]
```diff
@@ -142,7 +143,7 @@
 			"@ci:php:stan"
 		],
 		"ci:php:cs-fixer": "php-cs-fixer fix --config .php-cs-fixer.php -v --dry-run --diff",
-		"ci:php:lint": "find .*.php *.php Classes Configuration Tests -name '*.php' -print0 | xargs -r -0 -n 1 -P 4 php -l",
+		"ci:php:lint": ".Build/bin/parallel-lint --exclude .git --exclude .cache --exclude .Build --exclude typo3temp --exclude Documentation --exclude Resources --exclude .gitlab --exclude .github .",
```
Comment (oliverklee) [inline composer.json:146]:
```suggestion
		"ci:php:lint": "parallel-lint --exclude .git --exclude .cache --exclude .Build --exclude typo3temp --exclude Documentation --exclude Resources --exclude .gitlab --exclude .github .",
```

**Review by oliverklee — CHANGES_REQUESTED** [review]
Please also adapt the PR title to match the new purpose of this PR.

Also, currently the documentation renders with warnings which we'll need to fix.

File: composer.json [hunk 2]
```diff
@@ -142,7 +143,7 @@
 			"@ci:php:stan"
 		],
 		"ci:php:cs-fixer": "php-cs-fixer fix --config .php-cs-fixer.php -v --dry-run --diff",
-		"ci:php:lint": "find .*.php *.php Classes Configuration Tests -name '*.php' -print0 | xargs -r -0 -n 1 -P 4 php -l",
+		"ci:php:lint": "parallel-lint --exclude .git --exclude .cache --exclude .Build --exclude typo3temp --exclude Documentation --exclude Resources --exclude .gitlab --exclude .github .",
```
Comment (oliverklee) [inline composer.json:146]: I propose we use allowlisting for the file and directories to lint instead of switching to blacklisting (as that would need adjustment every time a tool create a PHP file).

**Review by oliverklee — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (Starkmann) [reply composer.json:146]: parallel-lint does not have whitelist:
https://github.com/php-parallel-lint/PHP-Parallel-Lint?tab=readme-ov-file#command-line-options

Comment (Starkmann) [reply composer.json:146]: And combining multiple calls is not what we want, or?

Comment (oliverklee) [reply composer.json:146]: I understand this line from the documentation that the last part of the call is the directory to check. (I haven't tried this, and I also have not tried whether we could provide multiple directories here.)

File: Documentation/Development.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.. include:: /Includes.rst.txt
+
+.. _dependency-manager:
+
+==================
+Dependency manager
+==================
```
Comment (oliverklee) [inline Documentation/Development.rst:7]: This file should not be part of this PR.

**Review by oliverklee — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (oliverklee) [inline composer.json:146]:
```suggestion
		"ci:php:lint": "parallel-lint .*.php *.php Classes Configuration Tests",
```

**Review by oliverklee — APPROVED** [review]
